sklearn/neighbors/dist_metrics.pyx

```diff
@@ -459,9 +459,13 @@ cdef class SEuclideanDistance(DistanceMetric):
             d += tmp * tmp / self.vec_ptr[j]
         return d
 
+    cdef inline DTYPE_t rdist(self, DTYPE_t* x1, DTYPE_t* x2,
+                              ITYPE_t size) nogil except -1:
+        return self._rdist(x1, x2, size)
```

Comment:
But this is definitely uglier and makes the code harder to understand. I think it at least deserves a comment referencing this pull request

Reply:
The other way to make this work is to not have inline. The generated C code for any `rdist` and `dist` are not marked with `CYTHON_INLINE`. I suspect because the base class `DistanceMetric` does not mark `rdist` and `dist` as inline. `DistanceMetric` can not mark them as inline because it would make them "final".

TLDR: Remove inline everywhere since Cython does not generated code that inlines `rdist` and `dist` anyways.

Reply:
I am fine with the last options but please check with a short micro-benchmark.

Reply:
For future reference, the Cython docs state:

> Note that class-level cdef functions are handled via a virtual function table, so the compiler won’t be able to inline them in almost all cases.

https://cython.readthedocs.io/en/latest/src/userguide/pyrex_differences.html#cdef-inline